two additional remarks in documentation

Konfekt opened the pull request:
These two points did not seem immediately clear to everyone

Konfekt committed:
make user aware of losing all data in tempname() dir

Konfekt committed:
motivate additional shellescape() escaping

chrisbra commented on runtime/doc/builtin.txt:
```diff
@@ -8991,11 +8991,12 @@ shellescape({string} [, {special}])			*shellescape()*
+		The {special} argument adds additional escaping of keywords
+		used in Vim commands. When it is not omitted and a non-zero
+		number or a non-empty String (|non-zero-arg|), then special 
+		words such as "!", "%", "#" and "<cword>" (as listed in 
```
can you keep the `items` instead of `words` please?

Konfekt committed:
it's items not words

as pointed out by @chrisbra in https://github.com/vim/vim/pull/14770#discussion_r1600664148

Konfekt commented on runtime/doc/builtin.txt:
```diff
@@ -10242,7 +10243,8 @@ tempname()					*tempname()* *temp-file-name*
 			:let tmpfile = tempname()
 			:exe "redir > " .. tmpfile
-<		For Unix, the file will be in a private directory |tempfile|.
+<		For Unix, the file will be in a private directory |tempfile|
```
Somehow this comment was dropped in https://github.com/vim/vim/commit/5faeb60480c6efba5c0468c01275120b6ace5a09
Is this clear to everyone? I got caught off-guard once and kept wondering for a while.

chrisbra replied:
It didn't get dropped, I just added the missing trailing dot back and then there was no change anymore :)

Konfekt replied:
When looking at https://github.com/vim/vim/pull/14770/files there's also https://github.com/vim/vim/pull/14770/files#diff-ef8946d89919ad628b9ac65c49d3067e4e017b0980cac70eb25d832a59ef677dR10247 that was meant to warn against recursive deletion of everything inside the temp dir...

chrisbra replied:
hm, I might have missed to squash the third commit, sorry, let me fix it

chrisbra replied:
updated and further clarified here: 5cf5301e2847c5a1a380ae030c58825b5c6fe275
